src/DependencyInjection/Compiler/EnableAuthenticatorManagerPass.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sensio\Bundle\FrameworkExtraBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * Optimizes the container by removing unneeded listeners.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+class EnableAuthenticatorManagerPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container)
+    {          
+        if(!$container->has('sensio_framework_extra.security.listener')){
+            return;
+        }
+        
+        if(!$container->hasExtension('security')){
+            return;
+        }
+        
+        $securityConfig = $container->getExtensionConfig('security')[0];
```

Comment:
Not sure about the [0] here ? I didn't find why getExtensionConfig() returns an array of arrays.
Please review.

Reply:
I'm nore sure we should parse the extension config to fetch this piece of information.

@wouterj Can you have a look please?

Reply:
Yes, I don't think it's a good idea to depend on the SecurityBundle config here.

If I'm correct, we can completely remove the if statement on SecurityListener:35 and always use the variables from line 95-104.
This means that users will have to take care to update their expression to take null tokens into account. But that's only when they explicitly decided to switch to the new system. In Symfony core, we have not considered this a BC break. But I'll let @fabpot decide about that for this bundle.

Reply:
I agree with your idea, but I was wondering how important the if statement was.